Project Management Process.md

```diff
@@ -0,0 +1,287 @@
+# IPFS Project Management Process
+
+**Status: work in progress, not even a full draft yet**
+
+## Introduction
+TODO: describe the purpose of this document
+
+- this is an opt-in process, nobody in the community is *required* to use this process
+
+## Table on Contents
+- [TL;DR](#tldr)
+- [Projects](#projects)
+- [Roles](#roles)
+  - [Product Owners](#product-owners)
+  - [Project Leads](#project-leads)
+  - [Project Managers](#project-managers)
+- [Structure](#structure)
+  - [Overview](#overview)
+  - [Goals](#goals)
+  - [Backlog](#backlog)
+  - [Milestones](#milestones)
+  - [Project Roadmap](#project-roadmap)
+  - [Organization Roadmap](#organization-roadmap)
+- [Workflow](#flow)
+  - [Pipelines](#pipelines)
+    - [Stages](#stages)
+    - [Flow](#flow)
+    - [Example](#example)
+  - [Weekly Updates](#weekly-updates)
+  - [Roadmap Updates](#roadmap-updates)
+- [Implementation Guideline](#implementation-guideline)
+
+## TL;DR
+
+The IPFS project management process has the following **cadence**:
+
+Weekly Updates
+- once a week (2 weeks?)
+- discuss last week's progress
+- discuss the plans for next week
+
+Roadmap Updates
+- every 3 months
```

Comment:
Having thought about this a bit more, I think we should try to drive towards a process where a quarterly meeting doesn't have too significant meaning and rather the work (that would be done in such a meeting) would take place on much more regular rhythm, ie. the roadmap could be updated weekly. Having the roadmap update tied to a 3 month cycle is bad for variety of reasons (@jbenet articulated this well #somewhere else) and we need to be able to adjust faster and generally have more flexibility. There's also the logistical part that just doesn't work for an organization like this: timezones, physical locations, etc., so a better approach would be to do that work async and online.

What I think is still important is to review that roadmap together and have the project leads / projects discuss it to make sure dependencies are identified and people generally understand the whys of the organization roadmap a little better (in addition to their own roadmap). Like, main work of generating the milestones for each project should happen *before* an "official" meeting so that when a meeting is had, people and projects are already prepared. Not sure yet what/how exactly this works out, but I think we should have that mindset as we develop the process.